### PR TITLE
Fix and refactor `info`.

### PR DESCRIPTION
**Pull request by reitermarkus**
### Changes to the core
- [x] Followed [hacking.md](https://github.com/caskroom/homebrew-cask/blob/master/doc/development/hacking.md).

---

<details>
<summary>

sample output: `brew cask info tower` (newer version available than is installed)</summary>

``` text
tower 2.4.0-315-880ab34c
from https://github.com/caskroom/homebrew-cask/blob/master/Casks/tower.rb
==> Name
Tower
==> Homepage
https://www.git-tower.com/
==> Installation
/usr/local/Caskroom/tower/2.3.5-308-9f8ed9a3 (68B)
==> Artifacts
Tower.app (app)
/Applications/Tower.app/Contents/MacOS/gittower (binary)
==> Caveats
Cask tower installs files under "/usr/local". The presence of such
files can cause warnings when running "brew doctor", which is considered
to be a bug in Homebrew-Cask.

```

</details>

<details>
<summary>

sample output: `brew cask info hazel` (multiple versions installed)</summary>

``` text
hazel 4.0.5
from https://github.com/caskroom/homebrew-cask/blob/master/Casks/hazel.rb
==> Name
Hazel
==> Homepage
https://www.noodlesoft.com/hazel.php
==> Installation
/usr/local/Caskroom/hazel/4.0.3 (7 files, 277.1K)
/usr/local/Caskroom/hazel/4.0.5 (7 files, 277.1K)
==> Artifacts
Install Hazel.app/Contents/Resources/Hazel.prefPane (prefpane)
```

</details>

---

Fixes https://github.com/caskroom/homebrew-cask/issues/23565.

### CODE REVIEW

**Comment by reitermarkus** [comment]
@caskroom/maintainers, any opinions on the formatting?

**Comment by yurikoles** [comment]
@reitermarkus seems like you need to update tests.

File: lib/hbc/dsl/caveats.rb [hunk 1]
```diff
@@ -31,9 +31,9 @@ def files_in_usr_local
     localpath = "/usr/local"
     return unless Hbc.homebrew_prefix.to_s.downcase.start_with?(localpath)
     puts <<-EOS.undent
-    Cask #{@cask} installs files under "#{localpath}".  The presence of such
-    files can cause warnings when running "brew doctor", which is considered
-    to be a bug in Homebrew-Cask.
+      Cask #{@cask} installs files under "#{localpath}". The presence of such
+      files can cause warnings when running "brew doctor", which is considered
+      to be a bug in Homebrew-Cask.
```
Comment (yurikoles) [inline lib/hbc/dsl/caveats.rb]: What is the purpose of this whitespace-only changes?

Comment (reitermarkus) [inline lib/hbc/dsl/caveats.rb]: The indentation is just for uniform code-style. And of course there shouldn't be two spaces after a dot.

**Comment by reitermarkus** [comment]
@yurikoles, yes, the tests need to be changed. I just want to get some feedback on the formatting before, so I don't have to change them twice.

File: lib/hbc/cli/info.rb [hunk 2]
```diff
@@ -14,26 +18,36 @@ def self.help
...
-PURPOSE
+  def self.installed_info(cask)
+    ohai "Installation"
+    if cask.installed?
+      cask.versions.each do |version|
+        versioned_staged_path = cask.caskroom_path.join(version)
+        puts "#{versioned_staged_path} (#{versioned_staged_path.abv})"
+      end
+    else
+      puts "not installed"
```
Comment (jawshooah) [inline lib/hbc/cli/info.rb]: I'd prefer that this be capitalized: `Not installed`

File: lib/hbc/cli/info.rb [hunk 3]
```diff
@@ -14,26 +18,36 @@ def self.help
...
+
+  def self.homepage_info(cask)
+    return unless cask.homepage
+    ohai "Homepage"
+    puts formatted_url(cask.homepage)
+  end
+
+  def self.token_info(cask)
+    puts "#{cask.token} #{cask.version}"
+    puts "from #{formatted_url(github_info(cask))}" if github_info(cask)
```
Comment (jawshooah) [inline lib/hbc/cli/info.rb]: Same here: `From {{url}}`

File: lib/hbc/cli/info.rb [hunk 4]
```diff
@@ -14,26 +18,36 @@ def self.help
...
+  end
+
+  def self.name_info(cask)
+    ohai cask.name.size > 1 ? "Names" : "Name"
+    puts cask.name.empty? ? "none" : cask.name
```
Comment (jawshooah) [inline lib/hbc/cli/info.rb]: And again here: `None`

**Comment by jawshooah** [comment]
@reitermarkus Apart from capitalization, I'm fine with the changes 👍

**Pull request merged**